src/components/DepositWidget/Form.tsx

```diff
@@ -147,7 +149,31 @@ export const Form: React.FC<FormProps> = (props: FormProps) => {
       // const hasErrors = Object.keys(newErrors).some(key => !!newErrors[key])
       // return !hasErrors
     }
-  }, [amountInput, validatorActive])
+  }, [amountInput, decimals, errors, totalAmount, validatorActive])
```

Comment:
To be honest, all values were added by prettier plugin, using the react-hook rules.
Anything that's a state variable used inside the effect is put in the watchlist. One way to reduce this would be to move the function inside the effect.
Anyway, I remember Dmitry suggested to refactor this in a way that we don't need to apply the validation here, wasn't it?

As for the infinite loop, react-hooks are smart enough to not update the state if there are no changes https://reactjs.org/docs/hooks-faq.html#is-there-something-like-forceupdate

Reply:
Yes, probably this is a smell of a not good practice. @DmitryBespalov would you like to make a proposal for this? If we don't address now this, we should review it together with #101 (kind of related)

> As for the infinite loop, react-hooks are smart enough to not update the state if there are no changes
Yep, but I guess if we change the errors (for other validation), then it triggers the effect, and then in case of an error, it changes errors again. It's true that in the second update it will stop :)

Reply:
You're right. Infinite loop.
Also I'm concerned about using `totalAmount` and other BN values in any hook dependency as they are often recreated from string in the API, so new object each time, rerender each time.

But it can be addressed later when we have real contracts. Maybe caching BN creation or using `BN.toString()` in dependencies instead of raw BN.

Reply:
But I think it's good that it re-evaluates the validation if `totalAmount` changes. For example, when pooling the new balances every 30s. I don't see a huge issue with that.

What I'm confused is why the prettier makes the effect depend on ALL the vars that it reads. 

Ahh, I just realised about the part that concerned me (the errors dependency in that hook). It will disapear if we don't use it inside, we pass a function to `setErrors` instead of using `errors` inside.